src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -2162,6 +2162,42 @@ private int codePointX(Encoding enc, CodeRange codeRange, byte[] bytes, int p, i
         }
     }
 
+    @CoreMethod(names = "undump", taintFrom = 0)
+    @ImportStatic(StringGuards.class)
+    public abstract static class UndumpNode extends CoreMethodArrayArgumentsNode {
+        @Specialization(guards = "isAsciiCompatible(string)")
+        protected RubyString undumpAsciiCompatible(RubyString string,
+                @CachedLanguage RubyLanguage language,
+                @Cached AllocateHelperNode allocateHelperNode,
+                @Cached RopeNodes.MakeLeafRopeNode makeLeafRopeNode) {
+            // Taken from org.jruby.RubyString#undump
+            RopeBuilder outputBytes = StringSupport.undump(string.rope, getContext(), this);
+
+            final Rope rope = makeLeafRopeNode
+                    .executeMake(
+                            outputBytes.getBytes(),
+                            outputBytes.getEncoding(),
+                            CR_UNKNOWN,
+                            outputBytes.getLength());
```

Comment:
This 4th argument is the character length, not the number of bytes.

It seems better to use MakeRopeNode to avoid dealing with this.